### ISSUE TITLE
Timeline editing issues

### ISSUE THREAD

**Issue by mafiesto4** [FlaxEngine/FlaxEngine#519]
Batched issues related to keyframes and curves editors used in animation timelines. Reported by @W2Wizard 

- [x] 1) Zooming into the timeline (of a scene animation) doesn't really have a limit, editor starts to freakout when you keep doing it and is laggy (fix: 347eaa9d075adbf5921f268731a283d5f721195f and 052366be66cb5306195f30fc673f7f22b1371668).
![image](https://user-images.githubusercontent.com/3464853/118132722-4e349280-b400-11eb-8e4e-6e8d9fc4aed4.png)

- [x] 2) The handle bars don't move accurately where the cursor is (fix: 9464c509fa4c9d2f278d9f22a34e6918bd5fdda6).

https://user-images.githubusercontent.com/3464853/118133004-96ec4b80-b400-11eb-9d1f-def1b2d8fdef.mp4

- [x] 3) Weird grey box (probably a keyframe knob). To reproduce just make something with a Vec3, make some keyframes and select one of them.

https://user-images.githubusercontent.com/3464853/118133060-a66b9480-b400-11eb-99f8-22af54a70793.mp4

- [x] 4) Clickable area for keyframes should be a bit bigger, right now you have to be pixel accurate.

https://user-images.githubusercontent.com/3464853/118133386-111cd000-b401-11eb-9822-e8373b1e618f.mp4

- [x] 5) The timeline view expandle would make it way more easier to see where your keyframes actually are (fix: bbe9193686c9f1f77cd35cb3060f36ed15ef3a2b).

![image](https://user-images.githubusercontent.com/3464853/118133439-21cd4600-b401-11eb-801d-ed62a7754179.png)

- [x] 6) Overall I personally prefer it way much when all of the keyframes are present in a single view instead, e.g UE4. Having it separate like that makes it imo harder to navigate. - we already have collapsed tracks with keyframes but it culd be improved for nested tracks + add ability to move keyframes from multiple tracks at once (rectangular selection). (added: d19d0ef0e598c83bc039ea898df5af52e3f031d3)

![image](https://user-images.githubusercontent.com/3464853/118133488-30b3f880-b401-11eb-9033-0c606932d06f.png)

**Comment by W2Wizard** [comment]
Another issue that I found is that overall the preview value gets really tiny and unreadable for Vec3 for instance once you reach 2 decimal places:

![image](https://user-images.githubusercontent.com/63303990/118257046-78448e00-b4ae-11eb-824a-dee86cbbc7c4.png)

**Comment by W2Wizard** [comment]
When having a nested timeline in you can add a reference to the same asset itself, when you do that, the editor freezes.

**Comment by W2Wizard** [comment]
When the camera track gets too small you get this:
![image](https://user-images.githubusercontent.com/63303990/118260004-184fe680-b4b2-11eb-946f-f04fa3af09af.png)

**Comment by W2Wizard** [comment]
- [ ] 10) Not even sure whats happening here, but the norm is usually:
`Scroll wheel` -> move up and down
`CTRL + Wheel` -> Zoom in and out

It seems for nested timelines this does not really work? Atleast for me that is, also the values seem to be `really` precise which the zoom in and out should avoid but it doesn't work. It just results in a really slow scrolling.

![image](https://user-images.githubusercontent.com/63303990/118273376-2149b380-b4c4-11eb-8b24-5cef84e57efc.png)

**Comment by W2Wizard** [comment]
- [ ] 11) Keyframe decides to go on a holiday

https://user-images.githubusercontent.com/63303990/118295778-83aeae00-b4dc-11eb-8d81-cba0878d96c4.mp4

**Comment by honzapatCZ** [comment]
-  [ ] 12) Multi track box selection(Why shouldnt I be able to select keyframes on mulitple tracks and move them at once)
-  [ ] 13) Scaling is completely fucked up - ie if you move out of the default scale, the keyframes do not move -> results in desynchronization
-  [ ] 14) Copy pasting - should be possible in same types, even cross track
-  [ ] 15) Bigger areas for selection(both moving(especially reordering tracks) and selecting(ie trying to select a keyframe))
-  [ ] 16) Creating new keyframe at point where one already is should replace previous keyframe
-  [ ] 17) Editor preview - the current is weird - you have to be in play mode, save asset, that makes the animation back to start and then rewind to where you were
-  [ ] 18) Double click to bring the value menu
-  [ ] 19) Folders or parents should proxy the keyframes beneath them.

**Comment by honzapatCZ** [comment]
-  [ ] 20) adding to the @W2Wizard scrolling, 
Shift + Scroll could move horizontally on the timeline 
Ctrl + Scoll could scale as is, but around the current time(or mouse position idk which one is better) in timeline, not the start

**Comment by honzapatCZ** [comment]
Audio Tracks, Camera tracks and SubScene tracks should behavely differently, rather than limiting the whole track for single asset(and only once), have them just limit to the use type, ie Camera for camera and let it take any cameras(yes you will loose the ability to tween camera's properties, but thats ok), similiar with audio(add copying of the track, stretching and most importantly not limit it to single audio file, let it be free, this way much less space will be wasted)

**Comment by mafiesto4** [comment]
-  [x] 22) Panning the timeline with a right-mouse button over background (added: 94b4c20b356e2df4f2a626072bdecb5914e1ef45)

**Comment by mafiesto4** [comment]
After a few weeks, all of those issues are done! If case of new feature requests or bugs for scene animations please report them separately.
Cut-Scenes editing is way more usable now.

![camera-cuts](https://user-images.githubusercontent.com/3464853/134474981-3743eb4a-d3b5-4ee0-9956-3c0156bff520.png)
